help function have wrong exit code.

When we want to check the help info for the test cases, we can see ERROR handling below.

Need to avoid such output for help info.

```
./check-alsabat.sh -h
Usage: ./check-alsabat.sh [OPTION]
    -s |  --sof-logger
        Open sof-logger trace the data will store at /home/pxl/sof-test/logs/check-alsabat/2020-09-24-14:44:03-14270
        Default Value: On    -p parameter |  --pcm_p parameter
        pcm for playback. Example: hw:0,0
    -n parameter |  --frames parameter
        test frames
        Default Value: 240000    -f parameter |  --frequency parameter
        target frequency
        Default Value: 997    -c parameter |  --pcm_c parameter
        pcm for capture. Example: hw:1,0
    -h |  --help
        this message

 Case Name: check alsabat
 Preconditions:
    This test case requires physical loopback between playback and capture.
    playback <=====>  capture
    nocodec : no need to use hw loopback cable, It support DSP loopback by quirk
 Description:
    run alsabat test on the specified pipelines
 Case step:
    1. Specify the pcm IDs for playback and catpure
    3. run alsabat test
 Expect result:
    The return value of alsabat is 0

2020-09-24 06:44:03 UTC [ERROR] Starting func_exit_handler(), exit status=2, FUNCNAME stack:
2020-09-24 06:44:03 UTC [ERROR]  _func_opt_dump_help()  @  /home/pxl/sof-test/case-lib/opt.sh
2020-09-24 06:44:03 UTC [ERROR]  func_opt_parse_option()  @  /home/pxl/sof-test/case-lib/opt.sh:121
2020-09-24 06:44:03 UTC [ERROR]  main()  @  ./check-alsabat.sh:40
2020-09-24 06:44:03 UTC [INFO] Test Result: N/A!

```

--- a/case-lib/opt.sh
+++ b/case-lib/opt.sh
@@ -90,6 +90,7 @@ func_opt_parse_option()
             printf '    -h |  --help\n'
             printf '\tthis message\n'
             _func_case_dump_descption
+            trap - EXIT
             exit 2
         }
 

